app/controllers/roles_controller.rb

```diff
@@ -72,8 +104,18 @@ def with_filter(options, &block)
     filter ? roles.member_of(filter) : roles
   end
     
-  def render_count(roles, options)
-    { count: roles.count } if options.has_key?(:count)
+  def render_count(dataset, options)
+    { count: dataset.count } if options.has_key?(:count)
   end
   
+  def render_results(dataset, order_by: nil, offset: nil, limit: nil)
```

Comment:
I like this code, but I'd _consider_ putting it in the model, which is where I'd expect the db interactions to go.  That said, I don't feel too strongly since it's pretty simple and likely won't be reused anywhere else.